### DISCUSSION TITLE
Plex - No luck with hardware transcoding

### DISCUSSION THREAD

**Discussion by Nedlinin** [tteck/Proxmox#329] [General]
Was trying to use the Plex script to setup an LXC with hardware transcoding enabled and haven't had any luck.

Machine is running on an N5105 processor which I've noted some people online complaining about similar issues seemingly stemming from Intel driver support.

I'm running on VE7.2 (installed and updated yesterday).

Any known workarounds or logs I can provide to help troubleshoot?

**Comment by tteck** [comment]
Google found this
https://chuckdickey.com/2022/03/plex-hardware-transcoding-with-a-jasper-lake-intel-celeron-n5105-cpu-on-ubuntu-server

**Reply by Nedlinin** [reply]
Yep.  So I'm new to Proxmox/LXCs in general but I had stumbled on that and tried it both in the host and the LXC with no luck; vainfo went from returning that it connected to the driver to unable to load the driver.

I was also able to verify using dmesg that the "guc" firmware failed to load (error code -101 if I remember correctly).

I also tried to update the LXC manually to 22.04 with no luck.  Then I recreated and manually updated to the newest intel drivers, again with no luck.

Finally I installed a 5.18 kernel on the host hoping for the best and.. nada.

Any idea if I should be trying these fixes directly on the Proxmox host versus the LXC?

**Reply by tteck** [reply]
only on the LXC

**Reply by Nedlinin** [reply]
So I blew it away and tried again using the stuff linked above and end up without hardware transcoding.

The dmesg log shows:


```
root@plex:/lib# dmesg | grep -iE "huc|guc|dmc"
[   14.904250] Setting dangerous option enable_guc - tainting kernel
[   14.930754] i915 0000:00:02.0: [drm] Finished loading DMC firmware i915/icl_dmc_ver1_09.bin (v1.9)
[   14.945888] i915 0000:00:02.0: Direct firmware load for i915/ehl_guc_69.0.3.bin failed with error -2
[   14.945925] i915 0000:00:02.0: [drm] GuC firmware i915/ehl_guc_69.0.3.bin: fetch failed with error -2
[   14.945938] i915 0000:00:02.0: [drm] GuC firmware(s) can be downloaded from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/i915
[   15.418761] i915 0000:00:02.0: [drm] GuC firmware i915/ehl_guc_69.0.3.bin version 0.0
[   15.418813] i915 0000:00:02.0: [drm] GuC is uninitialized
```

Which leads me to believe it is failing to download the file automatically; do you know where I would manually put this firmware file in order for it to be loaded if I download it as stated above?

**Comment by tteck** [comment]
You did create a privileged Plex LXC container?

**Reply by Nedlinin** [reply]
That is correct; I followed the defaults except for the disk space (bumped to 20GB) and the network I setup on a different bridge and with a static IP (networking is working fine).

**Reply by tteck** [reply]
May want to take a look here https://github.com/tteck/Proxmox/issues/8

**Reply by tteck** [reply]
Have you tried `apt-get install intel-opencl-icd`?

**Reply by Nedlinin** [reply]
I have the newest it shows

```
root@plex:~# apt-get install intel-opencl-icd
Reading package lists... Done
Building dependency tree       
Reading state information... Done
intel-opencl-icd is already the newest version (22.27.23681).
```


One thing I noticed from your link to #8 is the forum link step 1 states enable Intel Onboard Graphics *and* disable IOMMU.  I have IOMMU enabled as I'm using it for passthrough for some NICs to an opnsense install..

I might try to disable IOMMU later today and see if that makes the difference.

**Reply by hammondses** [reply]
> I have the newest it shows
> 
> ```
> root@plex:~# apt-get install intel-opencl-icd
> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> intel-opencl-icd is already the newest version (22.27.23681).
> ```
> 
> One thing I noticed from your link to #8 is the forum link step 1 states enable Intel Onboard Graphics _and_ disable IOMMU. I have IOMMU enabled as I'm using it for passthrough for some NICs to an opnsense install..
> 
> I might try to disable IOMMU later today and see if that makes the difference.

Did you ever resolves this? I've got the exact same issue, opnsense passthrough and no hw transcoding in plex.

Thanks!
